### PR TITLE
Setup Publishing Pipeline

### PR DESCRIPTION
**Pull request by christian512**
I added a pipeline to automatically commit the executed notebooks (generated by the tests) to the branch `gh-pages`. From there, they are automatically deployed to `qutip.github.io`. Also the publishing only occurs if all notebooks passed the tests.

### CODE REVIEW

File: .github/workflows/notebook_ci.yaml [hunk 1]
```diff
@@ -34,17 +34,37 @@ jobs:
...
     - name: Create Notebook Artifact
       uses: actions/upload-artifact@v3
       with:
         name: executed-notebooks
-        path: tutorials/notebooks/*.ipynb
+        path: notebooks/*.ipynb
+
+  publish:
+    needs: pytests
+    runs-on: ubuntu-latest
```
Comment (hodgestar) [inline .github/workflows/notebook_ci.yaml:55]: Once this is working we likely also want a:
```
if: github.ref == 'refs/heads/main'
```
here.

File: .github/workflows/notebook_ci.yaml [hunk 2]
```diff
@@ -34,17 +34,37 @@ jobs:
...
+    steps:
+      - uses: actions/checkout@v3
```
Comment (hodgestar) [inline .github/workflows/notebook_ci.yaml:57]: Maybe with `checkout@v3` we need something like:
```
with:
  token: ${{ secrets.GH_TOKEN }}
```
to get the right permissions. Not sure.

**Comment by hodgestar** [comment]
I activated gh-pages for this repo and the notebook should appear at https://qutip.org/qutip-tutorials.

**Comment by BoxiLi** [comment]
Maybe this is not the right PR to mention this. I just want to comment that for Qcircuit plotting you can also refer to the installation guide https://qutip-qip.readthedocs.io/en/stable/installation.html#additional-software-for-plotting-circuits

File: .github/workflows/notebook_ci.yaml (hunk 1 above)
Comment (christian512) [reply .github/workflows/notebook_ci.yaml:55]: Yes, I removed this for testing with other branches.

**Comment by christian512** [comment]
> Maybe this is not the right PR to mention this. I just want to comment that for Qcircuit plotting you can also refer to the installation guide https://qutip-qip.readthedocs.io/en/stable/installation.html#additional-software-for-plotting-circuits

I will add the software to the Action Setup once we have notebooks requiring that installation. I will also refer to that in the Readme. Thanks!

**Review by hodgestar — APPROVED** [review]

**Pull request merged**